llama-index-integrations/vector_stores/llama-index-vector-stores-azurecosmosnosql/llama_index/vector_stores/azurecosmosnosql/base.py

```diff
@@ -0,0 +1,246 @@
+            indexing_policy=self._indexing_policy,
+            default_ttl=self._cosmos_container_properties.get("default_ttl"),
+            offer_throughput=self._cosmos_container_properties.get("offer_throughput"),
+            unique_key_policy=self._cosmos_container_properties.get("unique_key_policy"),
+            conflict_resolution_policy=self._cosmos_container_properties.get("conflict_resolution_policy"),
+            analytical_storage_ttl=self._cosmos_container_properties.get("analytical_storage_ttl"),
+            computed_properties=self._cosmos_container_properties.get("computed_properties"),
+            etag=self._cosmos_container_properties.get("etag"),
+            match_condition=self._cosmos_container_properties.get("match_condition"),
+            session_token=self._cosmos_container_properties.get("session_token"),
+            initial_headers=self._cosmos_container_properties.get("initial_headers"),
+            vector_embedding_policy=self._vector_embedding_policy,
+        )
+
+    def add(
+            self,
+            nodes: List[BaseNode],
+            **add_kwargs: Any,
+    ) -> List[str]:
+        """Add nodes to index.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+
+        Returns:
+            A List of ids for successfully added nodes.
+
+        """
+        ids = []
+        data_to_insert = []
+
+        if not nodes:
+            raise Exception("Texts can not be null or empty")
+
+        for node in nodes:
+            metadata = node_to_metadata_dict(
+                node, remove_text=True, flat_metadata=self.flat_metadata
+            )
+
+            entry = {
+                self._id_key: node.node_id,
+                self._embedding_key: node.get_embedding(),
+                self._text_key: node.get_content(metadata_mode=MetadataMode.NONE) or "",
+                self._metadata_key: metadata,
+            }
+            data_to_insert.append(entry)
+            ids.append(node.node_id)
+
+        for item in data_to_insert:
+            self._container.upsert_item(item)
+
+        return ids
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+
+        if ref_doc_id is None:
+            raise ValueError("No id provided to delete.")
+
+        self._container.delete_item(ref_doc_id,partition_key=ref_doc_id)
+
+    @property
+    def client(self) -> Any:
+        """Return CosmosDB client."""
+        return self._cosmos_client
+
+    def _query(self, query: VectorStoreQuery) -> VectorStoreQueryResult:
+        params: Dict[str, Any] = {
+            "vector": query.query_embedding,
+            "path": self._embedding_key,
+            "k": query.similarity_top_k,
+        }
+
+        top_k_nodes = []
+        top_k_ids = []
+        top_k_scores = []
+
+        params["k"] = 1
```

Comment:
this will overwrite `query.similarity_top_k`, is it by design?

Reply:
Removed that piece of code.